parsons/scytl/scytl.py

```diff
@@ -0,0 +1,664 @@
+            print('HTTPError: {} {}'.format(e.code, zipfile_url))
+
+    def _get_latest_counties_scytl_info(
+        self, state: str, election_id: str, version_num: str
+    ) -> dict[str, CountyDetails]:
+        """
+        Fetch the settings JSON file for the election and parse the county details
+        for participating counties in a state election.
+
+        `Args`:
+            state: str
+                The two-letter state code for the state
+            election_id: str
+                The election ID for the given election
+            version_num: str
+                The latest version ID of the election as a string
+        `Returns`:
+            dict[str, CountyDetails]
+            A dictionary mapping county names to their sub-election information
+        """
+
+        county_dict = {}
+
+        config_settings_json_url = detail_settings_json_url_template.format(
+            state=state, election_id=election_id, version_num=version_num
+        )
+
+        settings_json_res = requests.get(config_settings_json_url)
+        settings_json = settings_json_res.json()
+
+        participating_counties = \
+            settings_json['settings']['electiondetails']['participatingcounties']
+
+        for county_row in participating_counties:
+            county_info = county_row.split('|')
+            source_county_name = county_info[0]
+            county_election_id = county_info[1]
+            county_version_num = county_info[2]
+            county_update_date = self._parse_date_to_utc(county_info[3])
+
+            county_details = CountyDetails(
+                state,
+                source_county_name,
+                county_election_id,
+                county_version_num,
+                county_update_date
+            )
+
+            county_dict[source_county_name] = county_details
+
+        return county_dict
+
+    def _parse_county_xml_data_to_precincts(
+        self, county_data: bytes, county_details: CountyDetails
+    ) -> list[object]:
+        """
+        Parse a detail XML file for a county into a list of election
+        results by precinct and vote method.
+
+        `Args`:
+            county_data: bytes
+                The detail XML file for a county as bytes
+            county_details: str
+                The details object for the county, including name,
+                id, and last updated datetime
+        `Returns`:
+            list[object]
+            The list of election results by precinct and vote method in the file.
+        """
+
+        tree = ET.fromstring(county_data)
+
+        precinct_dict = {}
+        precinct_votes = []
+
+        root = tree
+
+        for child in root:
+
+            if child.tag == 'VoterTurnout':
+                precincts = child[0]
+
+                for precinct in precincts:
+                    data = (precinct.attrib)
+                    name = data.get('name')
+
+                    precinct_info = {
```

Comment:
It looks like a lot of precinct info doesn't make it to the final function return. Am I missing something?

Reply:
Good catch, might as well add the rest to the dictionary.